src/Spice86.Core/Emulator/OperatingSystem/LocalFileSearchManager.cs

```diff
@@ -0,0 +1,134 @@
+            return false;
+        }
+        // the original code tolerated extra chars in pattern name beyond 8 unless it's a non-'*' check byte; our fixed-size truncation mirrors that behavior
+
+        // ---- EXT compare (early '*' accept) ----
+        return CompareSegment(fileExt, wildExt, DosExtlength, true) switch {
+            true => true,
+            false => false,
+            // If wild ext has a 4th char, and it's not '*', reject (DOSBox-like behavior)
+            _ => wildExtLength <= DosExtlength || wildExt[DosExtlength] == '*'
+        };
+    }
+
+    // Common 8.3 segment compare:
+    // - Compares up to 'length' characters.
+    // - '?' matches any char (including space padding).
+    // - If '*' is encountered:
+    //     - when earlyStarAcceptsTrue == true (extension), return true immediately;
+    //     - otherwise (name), treat as "stop comparing here" and return null (no decision).
+    // Returns:
+    //   true => definitively accept (only possible with earlyStarAcceptsTrue and '*' seen)
+    //   false => mismatch
+    //   null => matched the segment fully (or name-stopped at '*'), no final decision
+    private static bool? CompareSegment(ReadOnlySpan<char> target, ReadOnlySpan<char> pattern, int length,
+        bool earlyStarAcceptsTrue) {
+        for (int i = 0; i < length; i++) {
+            char wc = pattern[i];
+            if (wc == '*') {
+                return earlyStarAcceptsTrue ? true : null;
+            }
+
+            if (wc != '?' && wc != target[i]) {
+                return false;
+            }
+        }
+
+        return null;
+    }
+
+    private static void SplitTo83(ReadOnlySpan<char> file, Span<char> targetFileName, Span<char> targetFileExt,
+        out int extLength) {
+        targetFileName.Fill(' ');
+        targetFileExt.Fill(' ');
+
+        int dotPos = file.LastIndexOf('.');
+        ReadOnlySpan<char> fileNameRaw = dotPos >= 0 ? file[..dotPos] : file;
+        ReadOnlySpan<char> fileExtRaw =
+            dotPos >= 0 && dotPos + 1 < file.Length ? file[(dotPos + 1)..] : ReadOnlySpan<char>.Empty;
+        ToUpperInto(fileNameRaw[..Math.Min(fileNameRaw.Length, targetFileName.Length)], targetFileName);
+        ToUpperInto(fileExtRaw[..Math.Min(fileExtRaw.Length, targetFileExt.Length)], targetFileExt);
+        extLength = fileExtRaw.Length; // actual (untruncated) length for the DOSBox-style 4th-char check
+    }
+
+    private static bool WildcardMatchesHiddenFile(ReadOnlySpan<char> fileName, ReadOnlySpan<char> wildcard) {
+        if (fileName.IsEmpty) {
+            return false;
+        }
+
+        if (wildcard.IndexOfAny('?', '*') < 0) {
+            return false;
+        }
+
+        return fileName.Length >= 5 && fileName[0] == '.' &&
+               !fileName.Equals(".", StringComparison.Ordinal) &&
+               !fileName.Equals("..", StringComparison.Ordinal);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I don't think this is a hot path. Can this attribute (which is only a suggestion to the JIT anyway) be removed ?